homeassistant/components/sun.py

```diff
@@ -74,57 +119,108 @@ def name(self):
     @property
     def state(self):
         """Return the state of the sun."""
-        if self.next_rising > self.next_setting:
+        if (getattr(self, STATE_ATTR_NEXT_RISING) >
+                getattr(self, STATE_ATTR_NEXT_SETTING)):
             return STATE_ABOVE_HORIZON
 
         return STATE_BELOW_HORIZON
 
     @property
     def state_attributes(self):
         """Return the state attributes of the sun."""
-        return {
-            STATE_ATTR_NEXT_DAWN: self.next_dawn.isoformat(),
-            STATE_ATTR_NEXT_DUSK: self.next_dusk.isoformat(),
-            STATE_ATTR_NEXT_MIDNIGHT: self.next_midnight.isoformat(),
-            STATE_ATTR_NEXT_NOON: self.next_noon.isoformat(),
-            STATE_ATTR_NEXT_RISING: self.next_rising.isoformat(),
-            STATE_ATTR_NEXT_SETTING: self.next_setting.isoformat(),
-            STATE_ATTR_ELEVATION: round(self.solar_elevation, 2),
-            STATE_ATTR_AZIMUTH: round(self.solar_azimuth, 2)
-        }
+        attrs = {}
+        for attr in [STATE_ATTR_NEXT_DAWN,
+                     STATE_ATTR_NEXT_DUSK,
+                     STATE_ATTR_NEXT_MIDNIGHT,
+                     STATE_ATTR_NEXT_NOON,
+                     STATE_ATTR_NEXT_RISING,
+                     STATE_ATTR_NEXT_SETTING,
+                     STATE_ATTR_SUNRISE,
+                     STATE_ATTR_SUNSET]:
+            if attr in self._monitored_condtions:
+                attrs[attr] = getattr(self, attr).isoformat()
+        for attr in [STATE_ATTR_ELEVATION,
+                     STATE_ATTR_AZIMUTH]:
+            if attr in self._monitored_condtions:
+                attrs[attr] = round(getattr(self, attr), 2)
+        for attr in [STATE_ATTR_DAYLIGHT,
+                     STATE_ATTR_PREV_DAYLIGHT,
+                     STATE_ATTR_NEXT_DAYLIGHT]:
+            if attr in self._monitored_condtions:
+                attrs[attr] = getattr(self, attr)
+        return attrs
```

Comment:
Keep it simple and remove `getattr` handling

Reply:
Yep, this will change per the other suggestions.